src/main/java/org/opensearch/knn/index/util/ModelInfoExtractor.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.util;
+
+import org.opensearch.knn.index.MethodComponentContext;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.indices.ModelMetadata;
+
+import static org.opensearch.knn.index.util.Faiss.FAISS_BINARY_INDEX_DESCRIPTION_PREFIX;
+
+/**
+ * ModelInfoExtractor provides functionality to extract information from the model metadata
+ */
+public class ModelInfoExtractor {
+    /**
+     * Extracts the index description from the model metadata
+     *
+     * @param modelMetadata model metadata
+     * @return index description
+     */
+    public static String getIndexDescription(ModelMetadata modelMetadata) {
+        MethodComponentContext methodComponentContext = modelMetadata.getMethodComponentContext();
+        VectorDataType vectorDataType = modelMetadata.getVectorDataType();
+        String indexDescription = methodComponentContext.getName().toUpperCase();
```

Comment:
Is this the index description? I dont think it is

Reply:
I updated the code to use vector data type. https://github.com/heemin32/k-NN/blob/binary-java/src/main/java/org/opensearch/knn/index/IndexUtil.java#L321

Reply:
Thanks Heemin, let me update

Reply:
Updated to use vector data type for IVF support